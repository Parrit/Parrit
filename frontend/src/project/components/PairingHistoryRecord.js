"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PairingHistoryRecord = void 0;
var react_1 = __importDefault(require("react"));
var moment_1 = __importDefault(require("moment"));
var PairingHistoryRecord = function (props) {
    var localPairingTime = moment_1.default.now();
    return (react_1.default.createElement("div", { className: "pairing-history-record" },
        react_1.default.createElement("div", { className: "pairing-history-record-clock" }),
        react_1.default.createElement("h3", { className: "pairing-time" }, localPairingTime),
        react_1.default.createElement("div", { className: "pairing-boards-with-people" }, props.pairingBoardsWithPeople.map(function (pairingBoardWithPeople, idx) {
            return (react_1.default.createElement("div", { key: idx, className: "pairing-board-with-people" },
                react_1.default.createElement("div", { className: "pairing-board-name" },
                    pairingBoardWithPeople.name,
                    ":"),
                pairingBoardWithPeople.people.map(function (person, idx) {
                    return (react_1.default.createElement("span", { key: idx, className: "person-name" },
                        person.name,
                        react_1.default.createElement("span", { className: "person-names-plus-sign" }, "+")));
                })));
        })),
        react_1.default.createElement("div", { className: "dotted-line" })));
};
exports.PairingHistoryRecord = PairingHistoryRecord;
exports.default = exports.PairingHistoryRecord;
