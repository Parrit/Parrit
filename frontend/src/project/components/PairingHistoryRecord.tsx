import moment from "moment";
import React, {useContext} from "react";
import {PairingArrangementDTO} from "../interfaces/PairingArrangementDTO";
import {ProjectContext} from "../ProjectContext";
import {TrashIcon} from "../../shared/components/TrashIcon";

interface Props {
  pairingArrangement: PairingArrangementDTO;
}

const PairingHistoryRecord: React.FC<Props> = (props) => {
  const { pairingArrangement } = props;
  const formattedDate = moment(pairingArrangement.pairingTime).format("MMMM Do YYYY, h:mm a");
  const { deletePairingArrangement } = useContext(ProjectContext);

  return (
    <div className="pairing-history-record">
      <div className="pairing-history-record-clock" />
      <h3 className="pairing-time">{formattedDate}</h3>

      <div className="pairing-boards-with-people">
        {pairingArrangement.pairingHistories.map((history) => {
          return (
            <div key={pairingArrangement.id} className="pairing-board-with-people">
              <div className="pairing-board-name">{history.pairingBoardName}:</div>
              {history.people.map((person) => {
                return (
                  <span key={person.id} className="person-name">
                    {person.name}
                    <span className="person-names-plus-sign">+</span>
                  </span>
                );
              })}
            </div>
          );
        })}
      </div>
        <div className="delete-pairing-arrangement"
             onClick={() => {deletePairingArrangement(pairingArrangement.id)}}>
            <TrashIcon/>
        </div>
      <div className="dotted-line" />
    </div>
  );
};

export default PairingHistoryRecord;
