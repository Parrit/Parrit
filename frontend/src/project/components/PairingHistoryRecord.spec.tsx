import * as React from 'react'
import * as timezoneMock from 'timezone-mock'
import PairingHistoryRecord from "./PairingHistoryRecord";
import * as renderer from 'react-test-renderer';

describe('<PairingHistoryRecord/>', () => {
    let pairingArrangement

    beforeAll(() => {
        timezoneMock.register('US/Pacific')
    })

    afterAll(() => {
        timezoneMock.unregister()
    })

    beforeEach(() => {
        pairingArrangement = {
            pairingTime: '2016-03-08T17:30:00.000+0000',
            pairingHistories: [
                {pairingBoardName: 'board1', people: [{name: 'Bob'}, {name: 'Billy'}]},
                {pairingBoardName: 'board2', people: [{name: 'Alice'}]}
            ]
        }
    })

    it('should render correctly', () => {
        const tree = renderer.create(<PairingHistoryRecord pairingArrangement={pairingArrangement}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})
