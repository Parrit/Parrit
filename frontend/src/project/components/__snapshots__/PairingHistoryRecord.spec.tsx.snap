// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PairingHistoryRecord/> should render correctly 1`] = `
<div
  className="pairing-history-record"
>
  <div
    className="pairing-history-record-clock"
  />
  <h3
    className="pairing-time"
  >
    March 8th 2016, 9:30 am
  </h3>
  <div
    className="pairing-boards-with-people"
  >
    <div
      className="pairing-board-with-people"
    >
      <div
        className="pairing-board-name"
      >
        board1
        :
      </div>
      <span
        className="person-name"
      >
        Bob
        <span
          className="person-names-plus-sign"
        >
          +
        </span>
      </span>
      <span
        className="person-name"
      >
        Billy
        <span
          className="person-names-plus-sign"
        >
          +
        </span>
      </span>
    </div>
    <div
      className="pairing-board-with-people"
    >
      <div
        className="pairing-board-name"
      >
        board2
        :
      </div>
      <span
        className="person-name"
      >
        Alice
        <span
          className="person-names-plus-sign"
        >
          +
        </span>
      </span>
    </div>
  </div>
  <div
    className="delete-pairing-arrangement"
    onClick={[Function]}
  >
    <svg
      version="1.1"
      viewBox="0 0 13 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        trash
      </title>
      <path
        d="M11.9786928,0.147592181 L9.08384804,0.546780752 L8.09419825,0 L4.36126536,0.495744203 L3.8585889,1.23053256 L0.728020854,1.64392861 C0.251810339,1.70641389 -0.0675453598,2.05953164 0.0122053452,2.4319605 C0.094779084,2.80356175 0.545582295,3.05391671 1.02091061,2.99005206 L12.271759,1.49426738 C12.7481459,1.43040272 13.0676781,1.07797467 12.9877509,0.706373418 C12.9050007,0.334082488 12.4541975,0.0844172099 11.9786928,0.147592181 L11.9786928,0.147592181 Z"
        id="Shape"
      />
      <path
        d="M1.77635684e-15,3 L1.59980426,16 L11.499979,16 L13,3 L1.77635684e-15,3 L1.77635684e-15,3 Z M3.59692415,14.6179356 C3.23886753,14.6496108 2.92206921,14.4032294 2.88753583,14.0667878 L1.97458231,5.06453567 C1.94077595,4.72826531 2.20250262,4.42897783 2.55946872,4.39644659 C2.91734359,4.36391534 3.23414191,4.61115282 3.26885704,4.94827927 L4.1816288,13.9507026 C4.21652569,14.286973 3.95389025,14.5862605 3.59692415,14.6179356 L3.59692415,14.6179356 Z M6.48173366,14.6415636 C6.12276826,14.6424197 5.83123383,14.3694997 5.83032506,14.0316883 L5.80233485,4.98800163 C5.8005173,4.6500191 6.09114296,4.37435958 6.4499266,4.37435958 C6.80871024,4.37264741 7.10042642,4.64625232 7.10224397,4.98440607 L7.13023418,14.0279216 C7.13132471,14.3660753 6.84160783,14.6405363 6.48173366,14.6415636 L6.48173366,14.6415636 Z M10.1880741,14.0546314 C10.1611744,14.3917579 9.84982873,14.6443031 9.49177211,14.6196478 C9.13389724,14.5949926 8.86490038,14.3010128 8.89198182,13.9638864 L9.60027962,4.94485493 C9.62717931,4.60858456 9.93852499,4.35501205 10.2962181,4.37966731 C10.6542747,4.40432257 10.9234533,4.69830232 10.8963719,5.03542877 L10.1880741,14.0546314 L10.1880741,14.0546314 Z"
        id="noun_3058_cc"
      />
    </svg>
  </div>
  <div
    className="dotted-line"
  />
</div>
`;
